x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/calculations/moving_average.tsx

```diff
@@ -0,0 +1,139 @@
+    values: { name },
+  });
+};
+
+export type MovingAverageIndexPatternColumn = FormattedIndexPatternColumn &
+  ReferenceBasedIndexPatternColumn & {
+    operationType: 'moving_average';
+    params: {
+      window: number;
+    };
+  };
+
+export const movingAverageOperation: OperationDefinition<
+  MovingAverageIndexPatternColumn,
+  'fullReference'
+> = {
+  type: 'moving_average',
+  priority: 2,
+  displayName: i18n.translate('xpack.lens.indexPattern.movingAverage', {
+    defaultMessage: 'Moving Average',
+  }),
+  input: 'fullReference',
+  selectionStyle: 'full',
+  requiredReferences: [
+    {
+      input: ['field'],
+      validateMetadata: (metadata) =>
+        metadata.dataType === 'number' && metadata.isBucketed === false,
+    },
+  ],
+  getPossibleOperation: () => {
+    return {
+      dataType: 'number',
+      isBucketed: false,
+      scale: 'ratio',
+    };
+  },
+  getDefaultLabel: () => {
+    return i18n.translate('xpack.lens.indexPattern.movingAverage', {
+      defaultMessage: 'Moving Average',
+    });
+  },
+  toExpression: (layer, columnId) => {
+    return dateBasedOperationToExpression(layer, columnId, 'moving_average', {
+      window: [(layer.columns[columnId] as MovingAverageIndexPatternColumn).params.window],
+    });
+  },
+  buildColumn: ({ referenceIds, previousColumn, layer }) => {
+    const metric = layer.columns[referenceIds[0]];
+    return {
+      label: ofName(metric.label),
+      dataType: 'number',
+      operationType: 'moving_average',
+      isBucketed: false,
+      scale: 'ratio',
+      references: referenceIds,
+      params:
+        previousColumn?.dataType === 'number' &&
+        previousColumn.params &&
+        'format' in previousColumn.params &&
+        previousColumn.params.format
+          ? { format: previousColumn.params.format, window: 5 }
+          : { window: 5 },
+    };
+  },
+  paramEditor: MovingAverageParamEditor,
+  isTransferable: (column, newIndexPattern) => {
+    return hasDateField(newIndexPattern);
+  },
+  getErrorMessage: (layer: IndexPatternLayer) => {
+    return checkForDateHistogram(layer);
+  },
+};
+
+function MovingAverageParamEditor({
+  state,
+  setState,
+  currentColumn,
+  layerId,
+}: ParamEditorProps<MovingAverageIndexPatternColumn>) {
+  const [inputValue, setInputValue] = useState(String(currentColumn.params.window));
+
+  useDebounceWithOptions(
+    () => {
+      if (inputValue === '') {
+        return;
+      }
+      const inputNumber = Number(inputValue);
+      setState(
+        updateColumnParam({
+          state,
+          layerId,
+          currentColumn,
+          paramName: 'window',
+          value: inputNumber,
+        })
+      );
+    },
+    { skipFirstRender: true },
+    256,
+    [inputValue]
+  );
+  return (
+    <EuiFormRow
+      label={i18n.translate('xpack.lens.indexPattern.movingAverage.window', {
+        defaultMessage: 'Window size',
+      })}
+      display="columnCompressed"
+      fullWidth
```

Comment:
This form input is a good candidate for help text. I would like to explain some of the comments that we've written in the definition of the window, such as how it's trailing and how it deals with missing values.

Reply:
Do we have a design for that yet?

Reply:
There is no final design, but the idea that @MichaelMarcialis wants to explore is a popover-based help text. Tracking that effort here: https://github.com/elastic/kibana/issues/81780